### PR TITLE
Add support for ansible-lint and yamllint as Github actions.

### PR DESCRIPTION
**Pull request by rjeffman**
This PR add support for running Yamllint, Ansible-lint and Flake8 as Github actions that will run on every push. It also adds yamllint and ansible-lint to Azure pipelines to be executed for every pull request.

A few existing playbooks needed to be fixed due to an ansible-lint issue.

This PR does not fixes any issues found in the existing playbooks, instead, it disables all found in playbooks during development. The issues should be fixed by other patches.

More details can be found on the individual commits.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+name: Run Linters
+on:
+  - push
```
Comment (seocam) [inline .github/workflows/lint.yml:4]: I think we need:
```
on:
  - push
  - pull_request
```
With that we can remove the lints from azure.

the m
  - pull_request

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          ANSIBLE_MODULE_UTILS: plugins/module_utils
+          ANSIBLE_LIBRARY: plugins/modules
+
+      - name: Run yaml-lint
+        uses: ibiqlik/action-yamllint@v1
+
+      - name: Run flake8
+        uses: julianwachholz/flake8-action@v1
+        with:
+          checkName: "Python Lint"
```
Comment (seocam) [inline .github/workflows/lint.yml:34]: I'd use "flake8" in the name. There is a bunch of different python lints available. In this case I'd say it's better to be more specific.

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  - push
+jobs:
+  linters:
+    name: Run Linters
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
```
Comment (seocam) [inline .github/workflows/lint.yml:13]: I think we are using 3.6 on azure tests. I'd either use 3.6 here or change it there to 3.8 as well.

File: setup.cfg [hunk 4]
```diff
@@ -22,7 +22,8 @@ data_files =
     /usr/share/ansible/roles/ipareplica = roles/ipareplica/*
 
 [flake8]
-extend-ignore = E203
+extend-ignore = E203,D100,D101,D102,D103,D105,D107
```
Comment (seocam) [inline setup.cfg:25]: I think that by default flake8 doesn't look at docstrings. Why do you need that? If we are using flake8-docstrings we don't need pydocstyle and vice-versa.

File: tests/azure/azure-pipelines.yml [hunk 5]
```diff
@@ -39,6 +39,12 @@ stages:
     - script: pydocstyle .
       displayName: Verify docstings
 
+    - script: yamllint .
+      displayName: Verify docstings
```
Comment (seocam) [inline tests/azure/azure-pipelines.yml:43]: I don't think we need to keep lints in azure but if we do the displayName of this and the next step are mistaken.

File: setup.cfg (hunk 4 above)
Comment (seocam) [reply setup.cfg:25]: Also you should either ignore (D212 or D213) and (D202 or D203).

File: .github/workflows/lint.yml (hunk 1 above)
Comment (rjeffman) [reply .github/workflows/lint.yml:4]: done.

File: .github/workflows/lint.yml (hunk 3 above)
Comment (rjeffman) [reply .github/workflows/lint.yml]: done.

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            playbooks/*/*.yml
+        env:
+          ANSIBLE_MODULE_UTILS: plugins/module_utils
+          ANSIBLE_LIBRARY: plugins/modules
+
+      - name: Run yaml-lint
+        uses: ibiqlik/action-yamllint@v1
+
+      - name: Run Python linters
+        uses: rjeffman/python-lint-action@master
```
Comment (seocam) [inline .github/workflows/lint.yml:33]: :eyes: :fire:

File: setup.cfg (hunk 4 above)
Comment (rjeffman) [reply setup.cfg]: Added almost the same configuration to flake8. I needed to add D400 and D401 to flake8 to obtain the same result as pydocstyle.

File: tests/azure/azure-pipelines.yml (hunk 5 above)
Comment (rjeffman) [reply tests/azure/azure-pipelines.yml]: Removed all lints from Azure (which removed a whole stage).

File: setup.cfg [hunk 7]
```diff
@@ -22,7 +22,8 @@ data_files =
     /usr/share/ansible/roles/ipareplica = roles/ipareplica/*
 
 [flake8]
-extend-ignore = E203
+extend-ignore = E203, D1, D212, D203, D400, D401
```
Comment (seocam) [inline setup.cfg:25]: Why did you need to ignore D400 and D401 as well?

Comment (rjeffman) [reply setup.cfg:25]: In my local setup, flake8 complains about D400 and D401 (I have flake8-docstrings 1.5.0 and pydocstyle 5.0.1), and pydocstyle didn't. I wanted to have the same output on both, and lead by pydocstyle.

File: .github/workflows/lint.yml (hunk 2 above)
Comment (rjeffman) [reply .github/workflows/lint.yml]: The whole Python Lint process have changed.
